app/models/direct_file_json_data.rb

```diff
@@ -68,8 +68,13 @@ def find_matching_json_dependent(dependent)
     return nil unless dependents.respond_to?(:find)
 
     dependents.find do |json_dependent|
-      # TODO: find match based on SSN
-      json_dependent["firstName"] == dependent.first_name
+      next unless json_dependent.present?
+
+      json_tin = json_dependent["tin"]&.tr("-", "")
+      xml_ssn = dependent.ssn
+
+      next unless json_tin && xml_ssn
+      json_tin == xml_ssn
```

Comment:
_just in case_ either json blob or xml does not have a `tin` value... I don't want to say it's a match for instance if both `json_tin` and `xml_ssn` are `nil`